services/ocl-loader/src/main/java/org/eclipse/osc/services/ocl/loader/Ocl.java

```diff
@@ -31,18 +35,41 @@ public class Ocl {
     private List<Storage> storage;
     private Console console;
 
+    private static ObjectMapper theMapper = new ObjectMapper();
+
+    /**
+     * an OCL object might be passed to different plugins for processing,
+     * in case any plugin want to change the property of Ocl, we should not change
+     * the original
+     * Object, we should change a deep copy. The method is for deep copy
+     * 
+     * @return
+     */
+    public Ocl deepCopy() {
```

Comment:
I see the point. The first intention was to have ocl kind of immutable object once loaded. The plug-ins could use their own ocl internal object. No problem for me though to use deep copy.

Reply:
I think the plug-ins not only use the immutable ocl object but also have to clone a new one every time we call the `startManagedService()`. Am I right?